### ISSUE TITLE
proot-distro ubuntu - Menu items

### ISSUE THREAD

**Issue by anonimo82** [termux/termux-x11#91]
Hello everyone.
I successfully installed proot-distro ubuntu and built zink, so I now have HardWare acceleration with Termux-X11, and I installed my Apps with proot-distro, now I have a sort of Ubuntu with HW acceleration and I can launch virtually any App from Terminal (Termux session, when doing "xhost +, proot-distro login ubuntu --shared-tmp, and setting variables (like DISPLAY)".
Now, I'd like to have my Apps in the Applications menu, so something which can "hook" my Apps from proot-distro and show them in the menu.
Is it possible at all?

I also have another, minor issue: when I'm not using MHL, I have a very small interface (I labor to see any text or GUI elements - buttons, menus, etc). Is it possible to lower the resolution to, say, 1280x720?

Thanks for any help.

**Comment by twaik** [comment]
1. Prooted environments do not have access to android components, including vulkan. Are you sure it is hardware accelerated?
2. What exactly do you mean by saying "Application Menu"? If you mean Android's launcher it is impossible to add applications due to Android restrictions.
3. At this stage it is impossible to change resolution. But I'm working on it.

**Comment by wmcbtech30** [comment]
> What exactly do you mean by saying "Application Menu"? If you mean Android's launcher it is impossible to add applications due to Android restrictions

Using `am` you could launch activities from Termux. though you need to manually create an `.desktop` file entry for it or a script only if classpath and android system partitions is propely set and mounted in proot distros

**Comment by anonimo82** [comment]
> 1. Prooted environments do not have access to android components, including vulkan. Are you sure it is hardware accelerated?
Yes, I built Zink from Tewmux-disabled (here on github) and I have it working in proot-distro with --shared-tmp
In example, I get a very smooth performance with Art Of Illusion

> 2. What exactly do you mean by saying "Application Menu"? If you mean Android's launcher it is impossible to add applications due to Android restrictions.
No, I mean the Window Manager's menu (I'm using FluxBox at the moment)

> 3. At this stage it is impossible to change resolution. But I'm working on it.
Ok, looking forward for this =)

**Comment by anonimo82** [comment]
> > What exactly do you mean by saying "Application Menu"? If you mean Android's launcher it is impossible to add applications due to Android restrictions
> 
> Using `am` you could launch activities from Termux. though you need to manually create an `.desktop` file entry for it or a script only if classpath and android system partitions is propely set and mounted in proot distros

Excuse me, I understood almost nothing of your post - it's not you, I'm just too dumb to fully understand

**Comment by wmcbtech30** [comment]
> > > What exactly do you mean by saying "Application Menu"? If you mean Android's launcher it is impossible to add applications due to Android restrictions
> > 
> > 
> > Using `am` you could launch activities from Termux. though you need to manually create an `.desktop` file entry for it or a script only if classpath and android system partitions is propely set and mounted in proot distros
> 
> Excuse me, I understood almost nothing of your post - it's not you, I'm just too dumb to fully understand

@anonimo82 well you could have elaborated more since apps can be referred to something

**Comment by anonimo82** [comment]
Yes, sorry. I being generic, because I installed some, not just one.
I were meaning any of them (eg, lmms).

**Comment by twaik** [comment]
> KGSL give you access to /dev/dri/card0 without special privileges

I do not think it can be true... Kernel and SELinux restrict this behaviour...

**Comment by Heasterian** [comment]
I think that bylaws could explain it better than me, but case is that Turnip with KGSL support can be loaded from userspace and works fine in proot. I set it up couple of times, first time around half year ago. https://www.reddit.com/r/termux/comments/p7i5kg/world_of_warcraft_wrath_of_the_lich_king_with/

**Comment by wormstest** [comment]
The Mesa patch is required to compile a normally working turnip zink into proot,termux-x11,xsdl

**Comment by anonimo82** [comment]
Excuse me for late reply, I've been busy with something else.
What is the "am" command, in detail? (note I'm doing a Google search right now)

**Comment by wmcbtech30** [comment]
Activity Manager http://adbcommand.com/adbshell/am

**Comment by Heasterian** [comment]
> The acceleration patches floating around (https://github.com/suhan-paradkar/tewmux-disabled/releases/tag/mesa-hw-v2) focus on getting things to work inside termux itself rather than a proot distro. And I'm not sure what they actually fix and why every dep has to be compiled from source.

They are making some Termux specific changes so Xwayland runs faster and changes default Mesa config. It's nothing connected to DRI3.

**Comment by twaik** [comment]
> Am I right assuming that DRI is simply not possible without root?

Yes.

**Comment by Trass3r** [comment]
True simply no access to /dev/dri/card0 or /dev/dri/render128, just /dev/kgsl3d0.

> Zink+turnip works fine indeed

Btw looks like that path is now defunct due to https://www.supergoodcode.com/big-marge/

**Comment by twaik** [comment]
A question is not related to `termux-x11` itself, moving to termux-packages.
